### ISSUE TITLE
poetry init - doesn't find or install a given dependency, even with the exact name

### ISSUE THREAD

**Issue by sfermigier** [python-poetry/poetry#3434]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Mac OS 10.13
- **Poetry version**: 1.1.4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: irrelevant

## Issue

Trying to create a new project with `poetry init`, I ask to add dependencies interactively and I get:

```
Search for package to add (or leave blank to continue): PyGithub
Found 20 packages matching PyGithub

Enter package # to add, or the complete package name if it is not listed:
 [0] pygithub33
 [1] pygithub3
 [2] pygithub3-intellisense
 [3] PyGithub-ReadOnly
 [4] pygithub-redux
 [5] PyGithub-requests
 [6] pygit-annek
 [7] gitshoes
 [8] pygithubutils
 [9] poebot
 > PyGithub
Value "PyGithub" is invalid
```

There is a `PyGithub` on PyPI: <https://pypi.org/project/PyGithub/> so I guess the behaviour is wrong.

1. PyGithub should be the first match (as it's the best match).
2. It should be possible to add PyGithub in anycase by forcing the package name.

**Comment by sinoroc** [comment]
Duplicate https://github.com/python-poetry/poetry/issues/312 ?

**Comment by sfermigier** [comment]
@sinoroc Thanks, I missed this one. But #312 is considered as a usability issue, in the case of this issue, I consider it as a bug: one can't achieve the desired outcome using the workflow proposed by `poetry init`. 

Of course it's possible to skip the *add dependencies interactively* phase and add the dependencies manually in `pyproject.toml` or by using `poetry add` so there is a workaround.

Still, it's not just a usability issue.

**Comment by glencairn** [comment]
Digging into this, I believe @sfermigier is right, this is a bug. In fact, it looks like `poetry` actually fails twice to handle the situation described above.

1. After searching for packages that match the given name, in this case `PyGithub`, it tries to find an [_exact match_ from the search results](https://github.com/python-poetry/poetry/blob/master/poetry/console/commands/init.py#L258). However, the name comparison is _case-sensitive_ which goes against [PEP-426](https://www.python.org/dev/peps/pep-0426/#name).

   > All comparisons of distribution names MUST be case insensitive, and MUST consider hyphens and underscores to be equivalent.

2. The second point of failure is quite odd- when trying to build up the list of choices, `poetry` [explicitly chooses to _not_ include an exact match in the list](https://github.com/python-poetry/poetry/blob/master/poetry/console/commands/init.py#L268-L269). No clue why that's in there.

Anyways, the fix is pretty straight-forward and I can submit a PR for this.

---

Reference code: https://github.com/python-poetry/poetry/blob/master/poetry/console/commands/init.py#L252-L271

**Comment by mkniewallner** [comment]
It looks like this has been fixed by https://github.com/python-poetry/poetry/pull/5076, which was released in [1.2.0b1](https://github.com/python-poetry/poetry/releases/tag/1.2.0b1).

Tested on `1.2.0b1`:
```console
Search for package to add (or leave blank to continue): PyGithub
Found 20 packages matching PyGithub
Showing the first 10 matches

Enter package # to add, or the complete package name if it is not listed: 
 [0] PyGithub
 [1] pygithub33
 [2] pygithub3
 [3] pygithub3-intellisense
 [4] PyGithub-requests
 [5] PyGithub-ReadOnly
 [6] pygithub-redux
 [7] hdsr-pygithub
 [8] pygit-annek
 [9] gitshoes
 >
```

Also works on other variants like `PyGitHuB`, `PYGITHUB`, etc.
